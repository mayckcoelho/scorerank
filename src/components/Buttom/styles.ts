import styled from 'styled-components/native';

export const StyledButtom = styled.TouchableOpacity`
    margin: 40px 0 10px;
    border: 0;
    border-radius: 4px;
    padding: 16px;
    align-items: center;
    justify-content: center;

    flex-direction: row;
`

export const ButtomText = styled.Text`
    font-weight: bold;
    color: #fff;
    font-size: 15px;
`