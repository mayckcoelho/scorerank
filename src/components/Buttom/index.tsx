import React from 'react';
import { TouchableOpacityProps } from 'react-native';

import { StyledButtom, ButtomText } from './styles';
import Icon from "react-native-vector-icons/FontAwesome";
Icon.loadFont();

interface Props {
    text: string;
    color: string;
};

type ButtomProps = TouchableOpacityProps & Props;

const Buttom: React.FC<ButtomProps> = ({ text, color, onPress }) => {
    return (
        <StyledButtom style={{ backgroundColor: color }} onPress={onPress}>
            <ButtomText>{text}</ButtomText>
        </StyledButtom>
    );
}

export default Buttom;