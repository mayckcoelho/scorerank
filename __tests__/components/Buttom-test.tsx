import React from 'react';
import Buttom from '../../src/components/Buttom';
import renderer from 'react-test-renderer';

describe('Buttom should render correctly', () => {
    it('match snapshot', () => {
        const buttom = renderer
            .create(<Buttom text="Teste" color="#fff" />)
            .toJSON();
        expect(buttom).toMatchSnapshot();
    });
})