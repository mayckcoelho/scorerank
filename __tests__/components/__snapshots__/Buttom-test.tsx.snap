// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttom should render correctly match snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "black",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 40,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Teste
  </Text>
</View>
`;
